pal/inc/pal_mstypes.h

```diff
@@ -200,7 +200,7 @@ extern "C" {
 // they must be either signed or unsigned) and we want to be able to use 
 // __int64 as though it were intrinsic
 
-#ifdef BIT64
+#ifndef BIT64
```

Comment:
> I'm not using CMake, ported vcprox and sln files to gyp / xcodeproj 
> 
> BTW; IMO this issue is NOT whether BIT64 is defined or not.

The "pal" source (borrowing from [dotnet/coreclr](https://github.com/dotnet/coreclr)) has dependencies on a bunch of defs created through cmake. If you use gyp (thanks for doing that!!), need to make sure those defs are available, otherwise you may run into many problems in pal.

Since dotnet/coreclr supports __APPLE__, pal should work on apple platform. You just need to satisfy its dependencies.

The _Is_integral issue won't repro if both BIT64 and PLATFORM_UNIX are defined. See line 307.